### PR TITLE
Sandrine's Movie App

### PR DESCRIPTION
**Pull request by MindstormingAB**
Link to my deployed project: https://sandrines-movies-project.netlify.app/

My reflections on how this week's project turned out: It has been fun and challenging to enter deeply into APIs, components and React state and start developing multi-page applications.

Things I'd like to have clarified or explained in more detail: I'd like to understand where to put `// eslint-disable-next-line react-hooks/exhaustive-deps` in order to get rid of the warning message 'React Hook useEffect has a missing dependency'.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+The purpose of this project was to fetch a list of movies from [themoviedb.org](http://themoviedb.org/), display them on a page and  link to a movie detail page when clicking on a movie. The design on https://technigo-movies-project.netlify.com/ was to be followed.
 
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+This project made me use React router to create multi-page applications and pratice using APIs in React combinining `useState` with `useEffect`
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+After signing up for an API key on [themoviedb.org](http://themoviedb.org/), I studied the structure of the json in Postman.
+I first focused on fetching data from the "popular movies API" in the component MovieList.js and displayed it the component MovieCard.js.
```
Comment (karanmann) [inline README.md:10]: Its so nice to read the detailed readme. I really need to get better at this.

File: code/public/index.html [hunk 2]
```diff
@@ -13,13 +13,14 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>Technigo React App</title>
-  </head>
+  <title>Sandrine's Movie App</title>
+  <link rel="shortcut icon" href="%PUBLIC_URL%/icons/clapperboard.ico">
```
Comment (karanmann) [inline code/public/index.html:17]: That makes sense to me now. I was trying to figure this out.

File: code/src/App.js [hunk 3]
```diff
@@ -1,9 +1,35 @@
...
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
-  )
-}
+    <BrowserRouter>
+      <Header api={api} onApiChange={handleApiChange} />
+      <Switch>
+        <Route path='/' exact>
```
Comment (karanmann) [inline code/src/App.js:21]: Its great that you added the 404 page that shows up when there is an error.

File: code/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+const Header = ({ api, onApiChange }) => {
+  return (
+    <header>
```
Comment (karanmann) [inline code/src/components/Header.js:5]: This is great. For sure something I should add onto when I do my code next time.

File: code/src/components/MovieList.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    if (api === 'POPULAR_URL') fetchMovies();
+    if (api === 'TOP_URL') fetchTopMovies();
+    if (api === 'UPCOMING_URL') fetchUpcomingMovies();
+  }, [api]);
+
+  useEffect(() => {
+    setLoading(false);
+  }, [movies]);
+
+  if (loading) {
```
Comment (karanmann) [inline code/src/components/MovieList.js:51]: Great that you added the loading animation.

File: code/src/pages/NotFound.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (karanmann) [inline code/src/pages/NotFound.js:1]: Loved this.

File: code/src/index.css [hunk 7]
```diff
@@ -1,13 +1,225 @@
...
+  display: inline-flex;
+  color: #fff;
+  align-items: center;
+  font-weight: bold;
+  text-decoration: none;
+  left: 50px;
+  top: 50px;
+}
+
+.detail-page .back-link svg {
```
Comment (karanmann) [inline code/src/index.css:119]: I should have used this approach instead of making the entire back button. This is great :)

**Review by karanmann — APPROVED** [review]
Congratulations Sandrine, This is a solid and stong project. It was very clear to undertand and follow. I think I will be refering to your project when I shall be going back to mine.

File: code/src/App.js [hunk 8]
```diff
@@ -1,9 +1,35 @@
...
+import MovieList from './components/MovieList';
+import Footer from './components/Footer';
+import MovieDetails from './pages/MovieDetails';
+import NotFound from './pages/NotFound';
 
 export const App = () => {
+  const [api, setApi] = useState('POPULAR_URL');
+
+  const handleApiChange = event => {
+    setApi(event.target.value);
```
Comment (EmOlsz) [inline code/src/App.js:14]: Great ! 👍

File: code/src/components/MovieCard.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+const MovieCard = ({
+  poster_path,
+  title,
+  release_date,
+  id,
+}) => {
+  return (
+    <Link to={`/movies/${id}`}>
+      <img src={`https://image.tmdb.org/t/p/w342${poster_path}`} alt={title} ></img>
```
Comment (EmOlsz) [inline code/src/components/MovieCard.js:12]: `` is self-closing tag

File: code/src/pages/NotFound.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+import Arrow from '../icons/arrow.svg';
```
Comment (EmOlsz) [inline code/src/pages/NotFound.js:3]: Uppercase names are reserved for components only, not images

**Comment by MindstormingAB** [comment]
@EmOlsz Thank you for your feedback! I updated my code as requested.

**Pull request closed without merging**